chrome/content/zotero/xpcom/style.js

```diff
@@ -399,6 +431,72 @@ Zotero.Styles = new function() {
 			}
 		});
 	}
+	
+	/**
+	 * Populate menulist with locales
+	 */
+	this.populateLocaleList = function(menulist, prefLocale) {
+		if(!_initialized) this.init();
+		
+		var doc = menulist.ownerDocument;
+		var popup = doc.createElement('menupopup');
+		menulist.appendChild(popup);
+		
+		var desiredLocale = "";
+		var fallbackLocale = Zotero.locale;
+		
+		// Primary dialect conversion (e.g. "en" to "en-US")
+		if (Zotero.Styles.primaryDialects[prefLocale] !== undefined) {
+			prefLocale = Zotero.Styles.primaryDialects[prefLocale];
+		}
+		if (Zotero.Styles.primaryDialects[fallbackLocale] !== undefined) {
+			fallbackLocale = Zotero.Styles.primaryDialects[fallbackLocale];
+		}
+		
+		if (prefLocale) {
+			desiredLocale = prefLocale;
+		} else {
+			desiredLocale = fallbackLocale;
+		}
+		
+		var menuLocales = {};
+		var menuLocalesKeys = [];
+		var styleLocales = Zotero.Styles.locales;
+		
+		for (let locale in styleLocales) {
+			if (styleLocales[locale] !== undefined) {
```

Comment:
It is, and it's used in libraries that have to be compatible with unknown code, but since we're in a controlled environment, we don't worry about anything extending Object. (I'm not even positive anything still could with the sandboxing Mozilla does, but in any case we haven't had an issue with another extension doing that in 6 or 7 years, and they quickly fixed it, so avoiding unnecessary lines is more important.)